### PR TITLE
Vb 1705 create listener for receiving change of circumstances

### PR DESCRIPTION
**Pull request by AledWEvans**
User story

AS A developer

I WANT to be informed when there has been a change that may impact an existing booking

SO THAT the service can process the change

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/HmppsManagePrisonVisitsOrchestrationExceptionHandler.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.config
 
+import jakarta.validation.ValidationException
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/HmppsManagePrisonVisitsOrchestrationExceptionHandler.kt:3]: should this be javax.validation.ValidationException?

File: src/main/resources/application-dev.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+server:
```
Comment (dhirajshettymoj) [inline src/main/resources/application-dev.yml:1]: any changes needed to application.yml?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/HmppsManagePrisonVisitsOrchestrationExceptionHandler.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/config/HmppsManagePrisonVisitsOrchestrationExceptionHandler.kt:3]: As you mentioned javax package has been replaced with jakarta. Please ignore.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/EventFeatureSwitch.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service.listeners.events
+
+import org.springframework.core.env.Environment
+import org.springframework.stereotype.Component
+
+@Component
+class EventFeatureSwitch(private val environment: Environment) {
+  fun isEnabled(eventType: String): Boolean {
+    return environment.getProperty("feature.event.$eventType", Boolean::class.java, true)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/EventFeatureSwitch.kt:9]: As discussed, some tests to be added

File: src/main/resources/application-dev.yml (hunk 2 above)
Comment (AledWEvans) [reply src/main/resources/application-dev.yml:1]: not sure will double check

Comment (AledWEvans) [reply src/main/resources/application-dev.yml:1]: Looked at hmpps-prisoner-to-nomis-update project no config was required

Comment (AledWEvans) [reply src/main/resources/application-dev.yml:1]: apparently uses helm vars and follows a convention to map them over to env vars

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**